llumnix/backends/bladellm/llm_engine.py

```diff
@@ -0,0 +1,343 @@
+# Copyright (c) 2024, Alibaba Group;
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import traceback
+from typing import List, Optional, Tuple, Union, Iterable, Deque
+from collections import defaultdict
+import threading
+import asyncio
+import queue
+
+import ray
+from loguru import logger
+from ray.util.placement_group import PlacementGroup
+from ray.util.scheduling_strategies import PlacementGroupSchedulingStrategy, NodeAffinitySchedulingStrategy
+
+from blade_llm.service.engine import AsyncLLMEngine
+from blade_llm.service.args import ServingArgs
+from blade_llm.protocol import GenerateStreamResponse, ServerRequest
+from blade_llm.service.communications.engine_wrapper import APIWrapper
+from blade_llm.utils.disagg_utils import InstanceRole
+from blade_llm.service.disagg_pd_engine import PrefillAsyncLLMEngine, DecodeAsyncLLMEngine
+
+from llumnix.backends.backend_interface import BackendInterface, EngineState
+from llumnix.internal_config import MigrationConfig
+from llumnix.server_info import ServerInfo
+from llumnix.backends.utils import AsyncPutQueueActor
+from llumnix.llumlet.request import LlumnixRequest, RequestStatus
+from llumnix.instance_info import InstanceInfo
+from llumnix.queue.queue_type import QueueType
+
+class AsyncBackQueue(APIWrapper):
```

Comment:
![Uploading image.png…]()

Reply:
1. rename  "rpc" to "rayrpc".

2. rename AsyncBackQueueWrapper

Good Insight!